### PR TITLE
feat: add requested filters

### PR DESCRIPTION
**Pull request by sanchegm**
Description
* To filter via unfold, choices need to be in the model, not in admin. So I moved Status, Type, and CDWG choices into model.py and removed from admin.py
* Remove unused widget from admin.py.
* Add unfold filters to INSTALLED_APPS
* Type ignore in admin.py, similar to other unfold imports to avoid error.
* Add pylint disable=too-many-ancestors in model.py, To utilize model.py, we inherit from TextChoice. TextChoice triggered a pylint error because it has over the allowed limit of ancestors. From my understanding, I did not add additional ancestors.
* Unfold Django unfold dropdown filter [documentation](https://github.com/unfoldadmin/django-unfold?tab=readme-ov-file#dropdown-filters).
* Django Model field reference choices [documentation](https://docs.djangoproject.com/en/5.1/ref/models/fields/#choices).

Issue Ticket Number and Link  
Related to [#112](https://github.com/ClinGen/stanford-affils/issues/112)

Checklist  
[X ] I have reviewed the [how-to guide](https://github.com/ClinGen/stanford-affils/pull/how-to.md) and the [contributing file](https://github.com/ClinGen/stanford-affils/CONTRIBUTING.md).  
[X] I have run required [code checks](https://github.com/ClinGen/stanford-affils/pull/how-to.md#run-code-checks) and resolved any blockers.  
[X] I have created the necessary [tests](https://github.com/ClinGen/stanford-affils/src/app_test.py) to show my changes are effective and work as intended.  
[X] I have thoroughly commented my code, especially in more complex areas.  
[X] I have updated any necessary documentation.

Screenshots / Recordings Can select multiple filters for Status: 
Screenshot 2024-08-13 at 1 11 10 PM

Can select a single Type and CDWG: 
Screenshot 2024-08-13 at 1 11 13 PM
Screenshot 2024-08-13 at 1 11 19 PM

### CODE REVIEW

File: src/affiliations/models.py [hunk 1]
```diff
@@ -2,12 +2,58 @@
 
 # Third-party dependencies:
 from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class AffiliationStatus(models.TextChoices):  # pylint: disable=too-many-ancestors
+    """Creating choices for status."""
+
+    APPLYING = "APPLYING", _("Applying")
```
Comment (liammulh) [inline src/affiliations/models.py:11]: Out of curiosity, are these [tuples](https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences)? What is this `_("Some String")` syntax? I've never seen it before.

Comment (liammulh) [reply src/affiliations/models.py:11]: Apparently this is an internationalization thing: https://docs.python.org/3/library/gettext.html#class-based-api.

**Review by liammulh — APPROVED** [review]
Looks good!

File: src/affiliations/models.py (hunk 1 above)
Comment (sanchegm) [reply src/affiliations/models.py:11]: I followed the documentation listed here: https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types Essentially: ```Django supports adding an extra string value to the end of this tuple to be used as the human-readable name, or label. The label can be a lazy translatable string. ```

There are other ways to implement this without using the tuples/enumeration types, if you would prefer. Let me know if you'd like me to refactor this.

Comment (liammulh) [reply src/affiliations/models.py:11]: No need to refactor! I was just curious.

**Pull request merged**